### PR TITLE
Update Genplugin to support 1st and 3rd party plugins

### LINKED ISSUE #142: I want Gen Plugin to support 1st and 3rd party plugins.
As a plugin developer, I want Gen Plugin to support 1st and 3rd party plugins, so that I can generate the boilerplate for my plugins.

AC:
- [ ] 1st Party Plugins Generate:
  - [ ] \_\_init\_\_.py with the files contents that currently exist in `plugin.py.jinja2`
  - [ ] plugin_impl.py 
- [ ] 3rd Party Plugins Generate:
  - [ ] The existing templatesand plugin structure.
- [ ] 1st Party plugins can only be generated in a directory under `src/aac/plugins/` and the directory must only contain the plugin AaC model file.

### PR DESCRIPTION
**Pull request by Coffee2Bits**
Closes #142.

* Genplugin now has a second argument for first or third party plugin
* Fixed a template output bug; it will now generate **_multiple_** levels of directories
* Added some help command formatting so that it's less cramped/stacked
* Added a `choices` to the AaCCommand class so that plugins can provide a set of allowable choices
* Updated functional test workflow to test 1st and 3rd party plugin generation 
* Update GenPlugin model with new behavior
* Removed the check to have all AaC plugins prefixed with "aac"
* Updated the templates to be aligned with current implementation patterns.

### CODE REVIEW

File: .vscode/python.code-workspace [hunk 1]
```diff
@@ -18,7 +18,10 @@
         ],
         "python.testing.pytestEnabled": false,
         "python.testing.unittestEnabled": true,
-        "python.pythonPath": "python/venv/bin/python"
+        "python.pythonPath": "python/venv/bin/python",
+        "python.formatting.blackArgs": [
+            "--line-length=127"
```
[the remark this answers is not in the document]
Comment (Coffee2Bits) [reply .vscode/python.code-workspace:23]: I'm using the same value here as what's specified in tox.ini, and we should keep them synced.

Personally, I'm against any strict enforcement of line length; I think it should be up to the developer to maintain what's most readable just like they do for whitespace and the general construction of the code. However, having spent some time in reality, I agree with the obvious benefits of having a defined standard with enforcement for a consistent style like what we have here. Reducing the maximum line length to something like 99 is too aggressive in my opinion. I personally find it difficult to read aggressively shortened code, and I don't think that it adds enough value to accommodate some tools' gap in modern accessibility features like zoom or custom font size.

File: python/src/aac/templates/genplug/third_party/plugin_impl.py.jinja2 [hunk 2]
```diff
@@ -0,0 +1,39 @@
+"""AaC Plugin implementation module for the {{plugin.name}} plugin."""
+# NOTE: It is safe to edit this file.
+# This file is only initially generated by the aac gen-plugin, and it won't be overwritten if the file already exists.
```
[the remark this answers is not in the document]
Comment (Coffee2Bits) [reply python/src/aac/templates/genplug/third_party/plugin_impl.py.jinja2:3]: I think it sounds better with the "the" present.

[the remark this answers is not in the document]
Comment (cvchaparro) [reply python/src/aac/templates/genplug/third_party/plugin_impl.py.jinja2:3]: If we leave the "the" in, I think it makes more sense to say "... generated by the aac gen-plugin plugin ...".

**Comment by Coffee2Bits** [comment]
I liked Cameron's advice, so I changed the plugin a bit to automatically detect and generate first-party plugins.

Additional changes:
* Cleaned up the Gen-plugin plugin
* Removed the check to have all AaC plugins prefixed with "aac"
* Updated the templates to be aligned with current implementation patterns.

File: python/src/aac/plugins/gen_plugin/__init__.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def get_commands() -> list[AacCommand]:
         AacCommandArgument(
             "architecture_file",
             "The yaml file containing the AaC DSL of the plugin architecture.",
-        )
+        ),
```
Comment (Coffee2Bits) [inline python/src/aac/plugins/gen_plugin/__init__.py:24]:
```suggestion
        )
```

File: python/src/aac/templates/genplug/first_party/__init__.py.jinja2 [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    ]
+
+    return plugin_commands
+{%- if plugin_definitions | length > 0 -%}
+{# Leave a blank line here since we want 2 space if the expression is true and one if it's false#}
+{%- set yaml_file_joiner = joiner("\n---\n") %}
+
+
+@hookimpl
+def get_base_model_extensions() -> str:
```
Comment (cvchaparro) [inline python/src/aac/templates/genplug/first_party/__init__.py.jinja2:51]:
```suggestion
def get_plugin_aac_definitions() -> str:
```

File: python/src/aac/templates/genplug/first_party/plugin_impl.py.jinja2 [hunk 5]
```diff
@@ -31,9 +31,9 @@ def {{command.implementation_name}}({% for command_input in command.input -%}{{c
     {%- endif %}
     """
     # TODO add implementation here
-    def implement_me():
+    def _implement__and_rename_me():
```
Comment (cvchaparro) [inline python/src/aac/templates/genplug/first_party/plugin_impl.py.jinja2:34]:
```suggestion
    def _implement_and_rename_me():
```

**Review by cvchaparro — APPROVED** [review]
Overall looks good. Nice work.

**Pull request merged**